### PR TITLE
Update to openapi 2.9.2

### PR DESCRIPTION
**Pull request by laura-rodriguez**
* Update to 2.9.2 which was released during the development of this PR
* Add support for brands API (code + tests)
* Add support for OIE policies ((code + tests)
* Add support for Multipart FormData requests
* Implement upload files methods manually since the generator doesn't support this type of request, and we're soon migrating to OAS3.
* Mark as obsolete AssignRoleToGroup and AssignRoleToUser since we're adding a breaking change
* Fix a few Stylecop warnings

### CODE REVIEW

File: src/Okta.Sdk.IntegrationTests/NetworkZoneScenarios.cs [hunk 1]
```diff
@@ -126,7 +127,7 @@ private static NetworkZone BuildNetworkZoneObject()
             return new NetworkZone
             {
                 Type = NetworkZoneType.Ip,
-                Name = "newNetworkZone",
+                Name = "newNetworkZone" + Guid.NewGuid(),
```
Comment (andriizhegurov-okta) [inline src/Okta.Sdk.IntegrationTests/NetworkZoneScenarios.cs:130]: 👍

File: src/Okta.Sdk/Application.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    public partial class Application : IApplication
+    {
+        /// <inheritdoc/>
+        public string GetAccessPolicyId() =>
+
+            this.GetProperty<Resource>("_links")?
+                .GetProperty<Resource>("accessPolicy")?
+                .GetProperty<string>("href")?
+                .Split('/')?
+                .LastOrDefault() ?? null;
```
Comment (andriizhegurov-okta) [inline src/Okta.Sdk/Application.cs:20]: `?? null` can be omitted

File: src/Okta.Sdk.IntegrationTests/PoliciesScenarios.cs [hunk 3]
```diff
@@ -432,6 +462,158 @@ public async Task CreatePasswordPolicyRule()
...
+                GrantTypes = new List<OAuthGrantType>
+                {
+                    OAuthGrantType.Implicit,
+                    OAuthGrantType.AuthorizationCode,
+                },
+                ApplicationType = OpenIdConnectApplicationType.Web,
+            });
+
+            var accessPolicyId = createdApp.GetAccessPolicyId();
+            var accessPolicy = await client.Policies.GetPolicyAsync<AccessPolicy>(accessPolicyId);
```
Comment (andriizhegurov-okta) [inline src/Okta.Sdk.IntegrationTests/PoliciesScenarios.cs:501]: `accessPolicy` variable is not used further

File: src/Okta.Sdk/Group.cs [hunk 4]
```diff
@@ -23,5 +23,9 @@ public Task DeleteAsync(CancellationToken cancellationToken = default(Cancellati
         /// <inheritdoc/>
         public Task<IGroup> UpdateAsync(CancellationToken cancellationToken = default(CancellationToken))
             => GetClient().Groups.UpdateGroupAsync(this, Id, cancellationToken);
+
+        /// <inheritdoc/>
+        public Task<IRole> AssignRoleAsync(IAssignRoleRequest assignRoleRequest, string disableNotifications, CancellationToken cancellationToken = default(CancellationToken))
+            => GetClient().Groups.AssignRoleToGroupAsync(assignRoleRequest, Id, (bool?)null, cancellationToken);
```
Comment (andriizhegurov-okta) [inline src/Okta.Sdk/Group.cs:29]: `string disableNotifications` is not used in the function. Is this by intention?

File: src/Okta.Sdk/UsersClient.cs [hunk 5]
```diff
@@ -211,5 +211,10 @@ public Task DeactivateOrDeleteUserAsync(string userId)
         /// <inheritdoc />
         public Task<IUser> UpdateUserAsync(IUser user, string userId, CancellationToken cancellationToken = default)
             => UpdateUserAsync(user, userId, null, cancellationToken);
+
+        /// <inheritdoc />
+        public Task<IRole> AssignRoleToUserAsync(IAssignRoleRequest assignRoleRequest, string userId,
+            string disableNotifications, CancellationToken cancellationToken = default(CancellationToken))
+            => AssignRoleToUserAsync(assignRoleRequest, userId, (bool?)null, cancellationToken);
```
Comment (andriizhegurov-okta) [inline src/Okta.Sdk/UsersClient.cs:218]: Also here `string disableNotifications` is ignored. Is this made intentionally?

**Review by andriizhegurov-okta — APPROVED** [review]
LGTM with a few comments and questions.

**Review by bryanapellanes-okta — APPROVED** [review]
Minor nitpick comments.  LGTM 👍

File: src/Okta.Sdk/Application.cs (hunk 2 above)
Comment (laura-rodriguez) [reply src/Okta.Sdk/Application.cs:20]: Yes, I'll fix it.

File: src/Okta.Sdk/Group.cs (hunk 4 above)
Comment (laura-rodriguez) [reply src/Okta.Sdk/Group.cs:29]: The method with the string has been marked as obsolete and will be removed in the next major version. We're keeping it this way for backward compatibility.

File: src/Okta.Sdk.IntegrationTests/PoliciesScenarios.cs [hunk 6]
```diff
@@ -432,6 +462,158 @@ public async Task CreatePasswordPolicyRule()
...
+                },
+                GrantTypes = new List<OAuthGrantType>
+                {
+                    OAuthGrantType.Implicit,
+                    OAuthGrantType.AuthorizationCode,
+                },
+                ApplicationType = OpenIdConnectApplicationType.Web,
+            });
+
+            var accessPolicyId = createdApp.GetAccessPolicyId();
```
Comment (arvindkrishnakumar-okta) [inline src/Okta.Sdk.IntegrationTests/PoliciesScenarios.cs:500]: How is this id fetched from create app response? I see this interface defined in `IApplication`, but do not see the implementation. Also, I do not see a sample response that contains this Id in documentation at https://developer.okta.com/docs/reference/api/apps/#response-example-self-service-application-assignment-not-available.

File: src/Okta.Sdk.IntegrationTests/PoliciesScenarios.cs [hunk 7]
```diff
@@ -432,6 +462,156 @@ public async Task CreatePasswordPolicyRule()
...
+
```
Comment (arvindkrishnakumar-okta) [inline src/Okta.Sdk.IntegrationTests/PoliciesScenarios.cs:501]: Should we assert/verify for non null accessPolicyId here? If null, do not proceed with rest of the test?

File: src/Okta.Sdk.IntegrationTests/PoliciesScenarios.cs (hunk 6 above)
Comment (arvindkrishnakumar-okta) [reply src/Okta.Sdk.IntegrationTests/PoliciesScenarios.cs:500]: Got answered offline.

File: src/Okta.Sdk.IntegrationTests/PoliciesScenarios.cs (hunk 7 above)
Comment (laura-rodriguez) [reply src/Okta.Sdk.IntegrationTests/PoliciesScenarios.cs:501]: There's always an access policy by default when you create an application.

File: src/Okta.Sdk.IntegrationTests/PoliciesScenarios.cs [hunk 8]
```diff
@@ -432,6 +462,156 @@ public async Task CreatePasswordPolicyRule()
...
+            var defaultPolicyRule = await createdPolicy.ListPolicyRules().OfType<IProfileEnrollmentPolicyRule>().FirstOrDefaultAsync();
+            var profileAttributes = new List<IProfileEnrollmentPolicyRuleProfileAttribute>();
+            profileAttributes.Add(new ProfileEnrollmentPolicyRuleProfileAttribute
+            {
+                Name = "email",
+                Label = "Email",
+                Required = true,
+            });
+
+            defaultPolicyRule.Actions.ProfileEnrollment.Access = "ALLOW";
```
Comment (arvindkrishnakumar-okta) [inline src/Okta.Sdk.IntegrationTests/PoliciesScenarios.cs:565]: Actions, profile enrollment are being set here and the code segment that follows below (lines 575 to 589) is setting it again (twice).

Comment (laura-rodriguez) [reply src/Okta.Sdk.IntegrationTests/PoliciesScenarios.cs]: Good catch. Fixed.

**Pull request merged**